apis/projectcontour/v1alpha1/contourconfig.go

```diff
@@ -287,26 +361,59 @@ type EnvoyListenerConfig struct {
 	// default behavior in case of failures. Please file an issue if failures
 	// are encountered.
 	// See: https://github.com/projectcontour/contour/issues/3221
-	DisableAllowChunkedLength bool `json:"disableAllowChunkedLength"`
+	// +optional
+	DisableAllowChunkedLength bool `json:"disableAllowChunkedLength,omitempty"`
 
 	// ConnectionBalancer. If the value is exact, the listener will use the exact connection balancer
 	// See https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/listener.proto#envoy-api-msg-listener-connectionbalanceconfig
 	// for more information.
-	// +kubebuilder:validation:Enum="";"exact"
-	ConnectionBalancer string `json:"connectionBalancer"`
+	//
+	// Values: (empty string): use the default ConnectionBalancer, `exact`: use the Exact ConnectionBalancer
+	//
+	// Other values will produce an error.
+	// +optional
+	ConnectionBalancer string `json:"connectionBalancer,omitempty"`
 
 	// TLS holds various configurable Envoy TLS listener values.
-	TLS EnvoyTLS `json:"tls"`
+	// +optional
+	// +kubebuilder:default={minimumProtocolVersion: "1.2"}
```

Comment:
another that'll change over time, how to handle?

Reply:
This one is probably worth looking at the docstring again and making sure we explain that you *can* choose 1.1, but we don't turn it on by default.